tiledb/sm/filesystem/posix.cc

```diff
@@ -58,6 +58,18 @@ using filesystem::directory_entry;
 
 namespace tiledb::sm {
 
+struct UniqueDIRDeleter {
```

Comment:
You are using a deleter to implement a close method. It's confusing. The class to do this will not be complex at all.

Reply:
I have seen this pattern in other projects, and we are doing similar stuff in the C++ API but sure, I changed it to a dedicated class.

Reply:
Are you talking about the one in curl?

Reply:
Yes, and also [in the zstd compressor](https://github.com/TileDB-Inc/TileDB/blob/afa3c1dc709fb7dc765f27a36289b8a999d5fdd1/tiledb/sm/compressors/zstd_compressor.h#L67). Turns out I was wrong about the C++ API, we are using `shared_ptr` instead of `unique_ptr` there.

Reply:
The curl code is 5 years old... The ZSTD context one is a free function, which makes more sense as a deleter.